cmd/sops/common/common.go

```diff
@@ -139,6 +139,24 @@ func IsIniFile(path string) bool {
 	return strings.HasSuffix(path, ".ini")
 }
 
+// DefaultStoreForFormat returns the correct format-specific implementation
+// of the Store interface given the format string
+func DefaultStoreForFormat(format string) Store {
+	switch format {
+	case "yaml":
+		return &yaml.Store{}
+	case "json":
+		return &json.Store{}
+	case "dotenv":
+		return &dotenv.Store{}
+	case "ini":
+		return &ini.Store{}
+	case "binary":
+		return &json.BinaryStore{}
+	}
+	return &json.BinaryStore{}
```

Comment:
Instead of defaulting to `return &json.BinaryStore{}` here, I feel like it should fall back to `DefaultStoreForPath`, as that would keep with the current behavior.

Reply:
fixed in `DefaultStoreForPathOrFormat`